dist/bootstrap/index.js

```diff
@@ -5509,7 +5509,10 @@ function microk8s_init(addons) {
     return __awaiter(this, void 0, void 0, function* () {
         // microk8s needs some additional things done to ensure it's ready for Juju.
         // Add the given addons if any were given.
-        yield exec_as_microk8s("microk8s status --wait-ready");
+        if (!(yield exec_as_microk8s("microk8s status --wait-ready --timeout 900"))) {
+            core.setFailed("Timed out waiting for microk8s to be ready");
```

Comment:
Perhaps it was too presumptuous of me to assume that the timeout occurred causing this failure

```suggestion
            core.setFailed("microk8s couldn't become ready");
```